local-bd-etcd: fix CPMS

EmilienM opened the pull request:
* Re-use code from cpms.go
* Simplify error handling in cpms.go for errors out of getCPMS.
  It wasn't useful to count the number of CPMS, it'll always be zero
  when an error occurs.
* If an error is found when listing CPMS, just skip the test.
  We don't care how many CPMS are found, there is already a check
  in the function. This test can only function when one CPMS is found. OpenShift CPMS can only have 1 CPMS for now.
* Stop returning numCpms, we don't need to know at it should always
  be one, otherwise we return an error.
* Increase the size of the block device required for etcd. etcd grow to
  max 8GB, so let's have some window and require 10 minimum.

EmilienM commented:
/cc pierreprinetti mandre

EmilienM commented:
/uncc dulek eurijon
/cc itzikb-redhat

itzikb-redhat commented on test/extended/openstack/local-bd-etcd.go:
```diff
@@ -28,9 +28,33 @@ var _ = g.Describe("[sig-installer][Suite:openshift/openstack] The OpenShift clu
-	var clusterCPMS objx.Map
 	var clientSet *kubernetes.Clientset
 
+	getControlPlaneMachineSets := func(ctx context.Context, dc dynamic.Interface) ([]objx.Map, error) {
```
I have getControlPlaneMAchineSet at cpms.go https://github.com/openshift/openstack-test/pull/124/commits/cb55d18920ac8cd2724122e427f17709d6d45f35#diff-7b12ea0dbd387e5b0d5a4eef6d705422cd7eabd7cd8093be463d17d9eb882841R132 , Please take a look and tell me if you think it should be removed/moved etc

EmilienM replied:
excellent, I'll refactor :-)

itzikb-redhat commented:
/lgtm

EmilienM commented on test/extended/openstack/cpms.go:
```diff
@@ -43,13 +43,9 @@ var _ = g.Describe("[sig-installer][Suite:openshift/openstack] ControlPlane Mach
 		skipUnlessMachineAPIOperator(ctx, dc, clientSet.CoreV1().Namespaces())
 		controlPlaneMachineSet, numCpms, err = getControlPlaneMachineSet(ctx, dc)
 		if err != nil {
-			if numCpms == 0 {
```
this was not useful. in case of an error, `numCpms` will always be `0`. Let's just skip the test if CPMS can't be found (in case of UPI for example).

itzikb-redhat replied:
Is it possible that due to a bug there will be more than one CPMS?

MaysaMacedo commented on test/extended/openstack/cpms.go:
```diff
@@ -43,13 +43,9 @@ var _ = g.Describe("[sig-installer][Suite:openshift/openstack] ControlPlane Mach
-			} else {
-				o.Expect(err).NotTo(o.HaveOccurred())
-			}
+			e2eskipper.Skipf("Failed to get control plane machine sets: %v", err)
```
If any error happen we would skip the test, is this really desired?

EmilienM replied:
yes, we rely on CPMS data, so let's skip the test if we can't get it.

MaysaMacedo replied:
disregard my comment. I didn't see your last comment before I commented

MaysaMacedo commented:
/lgtm

let's see result of jobs

MaysaMacedo commented:
/lgtm
wait for CI before approval

MaysaMacedo commented:
/approve

All jobs are required anyway

EmilienM commented:
@pierreprinetti @itzikb-redhat See the change in `cpms.go`; I think I simplified it a bit. Also note that I made the change to cpms.go into a separate commit so this can be cherry-picked without the other file, since the etcd tests is only for 4.15 and beyond.

@MaysaMacedo I think I've fixed the error we discussed offline, we'll see how CI works now.

MaysaMacedo commented:
/lgtm
/hold for review of other folks if needed

EmilienM commented:
I tested locally with a deployment with etcd on additional disk and the test was skipped so something is wrong...

I'll look into it tomorrow. Sorry for the UPI job breakage in the meantime.

EmilienM commented:
```
passed: (6.5s) 2023-10-31T01:10:19 "[sig-installer][Suite:openshift/openstack] The OpenShift cluster runs with etcd on ephemeral local block device"
```
Got it working :-)

itzikb-redhat commented on test/extended/openstack/local-bd-etcd.go:
```diff
@@ -102,15 +78,20 @@ var _ = g.Describe("[sig-installer][Suite:openshift/openstack] The OpenShift clu
 		g.By("checking that the control plane flavor has enough ephemeral storage")
 		{
-			controlPlaneFlavor = clusterCPMS.Get("spec.template.spec.providerSpec.value.flavor").String()
+			controlPlaneFlavor = cpms.Get("spec.template.machines_v1beta1_machine_openshift_io.spec.providerSpec.value.flavor").String()
```
You may want to declare  the provider spec value spec.template.machines_v1beta1_machine_openshift_io.spec.providerSpec.value once

EmilienM replied:
done

EmilienM commented:
Tested with latest patchset:
```
passed: (5.9s) 2023-10-31T13:02:22 "[sig-installer][Suite:openshift/openstack] The OpenShift cluster runs with etcd on ephemeral local block device"
```

MaysaMacedo commented:
/lgtm

EmilienM commented:
/hold cancel
I think this is good to go, we can always iterate later.
@itzikb-redhat I'll let you cherry-pick the second commit to 4.14 if you want.
